### PR TITLE
feat: add documentation site

### PR DESCRIPTION
**Pull request by branchvincent**
This uses [mkdocs](https://www.mkdocs.org/) and [mkdocs-click](https://github.com/DataDog/mkdocs-click) to generate a documentation site (hosted for free by GitHub Pages) from simple markdown files.

Note, I also tried the popular [sphinx](https://www.sphinx-doc.org/en/master/) and [sphinx-click](https://github.com/click-contrib/sphinx-click), but I found it far more complex.

Preview:
image

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Documentation
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [master]
+    paths: ['docs/**']
```
Comment (mgagliardo91) [inline .github/workflows/docs.yaml:7]: it compiles docs from `click` too right? We should probably have it update on every push

**Review by mgagliardo91 — APPROVED** [review]

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (branchvincent) [reply .github/workflows/docs.yaml:7]: correct, maybe this should just be on release. otherwise, docs will show unreleased changes.

Comment (mgagliardo91) [reply .github/workflows/docs.yaml:7]: 👍 I think thats a good idea, can github version the docs?

Comment (branchvincent) [reply .github/workflows/docs.yaml:7]: looks like there's a [plugin](https://github.com/jimporter/mike) to mkdocs to allow gh-pages to host multiple versions. for now, i think just having docs for the latest release version is sufficient. the main documentation is for the cli, for which there's no reason to always use the latest version

**Pull request merged**